lib/crypto/crypto_compat.c

```diff
@@ -0,0 +1,147 @@
+#include <assert.h>
+
+#include <openssl/bn.h>
+#include <openssl/rsa.h>
+
+#include "crypto_compat.h"
+
+/**
+ * crypto_compat_RSA_valid_size(rsa):
+ * Check that the RSA ${rsa} has the correct size.
+ */
+int
+crypto_compat_RSA_valid_size(const RSA * const rsa)
+{
+
+	/* Sanity checks. */
+	assert(rsa != NULL);
+
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+	assert(rsa->n != NULL);
+	if ((RSA_size(rsa) != 256) || (BN_num_bits(rsa->n) != 2048))
```

Comment:
I thought that functions were supposed to have a single `return` statement?  Or is this function small enough to be an exception to that rule?